password_preload.js

```diff
@@ -11,6 +11,10 @@ const localeMessages = ipcRenderer.sendSync('locale-data');
 window.theme = config.theme;
 window.i18n = i18n.setup(locale, localeMessages);
 
+window.getEnvironment = () => config.environment;
+window.getVersion = () => config.version;
+window.getAppInstance = () => config.appInstance;
+
```

Comment:
Any reason you moved this up a few lines?

Reply:
Sorry should have said in description
This was causing the password_preload to fail to load, I assume from the last upstream merge that signal requires these globals available before you require it

Reply:
Good catch! Thanks